### PR TITLE
🎉 New Source: Confluence

### PR DESCRIPTION
**Pull request by tuanchris**
## What
- Add new source Confluence
- https://github.com/airbytehq/hacktoberfest-2021/issues/81
- Streams included: audit, blog_posts, group, pages, space, theme

## How
image
Stream ran successfully with basic normalization 

image
Local build successful with gradle

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [X] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [X] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [X] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]
LGTM just 2 small changes

**Review by harshithmullapudi — APPROVED** [review]
LGTM

**Comment by harshithmullapudi** [comment]
Waiting for #7249

**Comment by harshithmullapudi** [comment]
@tuanchris can you pull the latest source-acceptence image and test the integration tests ?

**Comment by tuanchris** [comment]
Hi @harshithmullapudi, Just did that 
image

File: airbyte-integrations/connectors/source-confluence/source_confluence/source.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+from abc import ABC
+from base64 import b64encode
+from typing import Any, Dict, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+from airbyte_cdk.sources.utils.transform import TransformConfig, TypeTransformer
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-confluence/source_confluence/source.py:15]: is this needed ?

Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-confluence/source_confluence/source.py:15]: I don't see it being used anywhere can you remove this

File: airbyte-integrations/connectors/source-confluence/source_confluence/source.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params.update({"type": self.content_type})
+        return params
+
+
+class Pages(BaseContentStream):
+    content_type = "page"
+
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-confluence/source_confluence/source.py:83]: Also do you mind adding docs url here for every stream ?

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]

**Comment by harshithmullapudi** [comment]
Also can you run `./gradlew format`

**Comment by tuanchris** [comment]
done @harshithmullapudi

File: airbyte-integrations/connectors/source-confluence/source_confluence/source.py (hunk 1 above)
Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-confluence/source_confluence/source.py:15]: ^ @tuanchris

File: airbyte-integrations/connectors/source-confluence/source_confluence/source.py (hunk 2 above)
Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-confluence/source_confluence/source.py:83]: ^ @tuanchris

**Review by harshithmullapudi — APPROVED** [review]
Once you are done with those small changes I can merge this and then publish the connector

File: airbyte-integrations/connectors/source-confluence/source_confluence/source.py (hunk 1 above)
Comment (tuanchris) [reply airbyte-integrations/connectors/source-confluence/source_confluence/source.py:15]: This is needed to normalize the data according to the schema. Removing this gives the following error: 
image

File: airbyte-integrations/connectors/source-confluence/source_confluence/source.py (hunk 2 above)
Comment (tuanchris) [reply airbyte-integrations/connectors/source-confluence/source_confluence/source.py]: I have added the documentation strings

**Comment by harshithmullapudi** [comment]
Thank you @tuanchris for this contribution

**Pull request merged**